core/src/test/java/de/evoila/cf/broker/controller/core/CatalogControllerTest.java

```diff
@@ -0,0 +1,136 @@
+    private static final String REQUEST_IDENTITY = "a82715af-3dcd-4650-bb90-dad096294e16";
+    private static final String ORIGINATING_IDENTITY = "cloudfoundry eyANCiAgInVzZXJfaWQiOiAiNjgzZWE3NDgtMzA5Mi00ZmY0LWI2NTYtMzljYWNjNGQ1MzYwIg0KfQ==";
+
+    @Mock
+    private Catalog catalog;
+    @Mock
+    private CatalogService catalogService;
+    private CatalogController catalogController;
+
+    @BeforeEach
+    void setUp() {
+        this.catalogController = new CatalogController(catalogService);
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+    /**
+     * Reads a JSON file containing an array of ServiceDefinitions and returns a list containing all these service
+     * definitions.
+     * @param filePath  The path (relative or absolute) to the JSON file.
+     * @return  A list containing all service definitions contained in the submitted file.
+     * @throws Exception
+     */
+    private List<ServiceDefinition> readServicesFile(String filePath) throws Exception
+    {
+        List<ServiceDefinition> generatedServices;
+        try (FileReader fileReader = new FileReader(new File(filePath))) {
+            generatedServices = new Gson().fromJson(fileReader, new TypeToken<List<ServiceDefinition>>() {}.getType());
+        }
+        return generatedServices;
+    }
+
+    /**
+     * A function testing CatalogController.getCatalog with a valid catalog.
+     * The catalog is stored in a JSON file.
+     * Not an @Test but called in @Test methods.
+     * @param catalogPath           The path of the JSON file containing the catalog.
+     * @param requestIdentity       A string representing a request identity.
+     * @param originatingIdentity   A string representing a originating identity.
+     * @throws Exception
+     */
+    private void validCatalogTest(String catalogPath,
+                                  String requestIdentity,
+                                  String originatingIdentity) throws Exception
+    {
+        // mocks
+        when(catalogService.getCatalog()).thenReturn(catalog);
+
+        List<ServiceDefinition> generatedServices = readServicesFile(catalogPath);
+        when(catalog.getServices()).thenReturn(generatedServices);
+        // actual method call
+        ResponseEntity<Catalog> response = catalogController.getCatalog(requestIdentity,
+                originatingIdentity);
+        // create expected object
+        Catalog expectedCatalog = new Catalog();
+        List<ServiceDefinition> expectedServices = readServicesFile(catalogPath);
+        expectedCatalog.setServices(expectedServices);
+        // check results
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(expectedCatalog, response.getBody()); // ist dieser Test wirklich sinnvoll? Wir mocken ja gerade getServices. Wäre es ausreichend zu prüfen, dass der Catalog nicht null ist? Oder nicht leer? In diesen Fällen bräuchte man auch die ganzen equals Funktionen nicht hinzufügen.
```

Comment:
But to answer your comment. This test is not that significant, since in the end it tests, whether the Controller does call the getter method of the catalog service correctly. It would be more effective, if the catalog would be generated by Spring, as it is the case in a full Spring application context buildup.

Reply:
In my opinion this would be tested when writing integration tests, what we currently want to achieve is testing the modules via simple unit tests and mock the Spring application around it.